framework/src/main/java/org/checkerframework/framework/util/CheckerMain.java

```diff
@@ -407,17 +409,65 @@ protected void addMainToArgs(final List<String> args) {
 
         if (!argsListHasClassPath(argListFiles)) {
             args.add("-classpath");
-            args.add(quote(String.join(File.pathSeparator, cpOpts)));
+            args.add(quote(concatenatePaths(cpOpts)));
         }
         if (!argsListHasProcessorPath(argListFiles)) {
             args.add("-processorpath");
-            args.add(quote(String.join(File.pathSeparator, ppOpts)));
+            args.add(quote(concatenatePaths(ppOpts)));
         }
 
         args.addAll(toolOpts);
         return args;
     }
 
+    /** Given a list of paths, concatenate them to form a single path. Also expand wildcards. */
+    private String concatenatePaths(List<String> paths) {
+        List<String> elements = new ArrayList<>();
+        for (String path : paths) {
+            for (String element : path.split(File.pathSeparator)) {
+                elements.addAll(expandWildcards(element));
+            }
+        }
+        return String.join(File.pathSeparator, elements);
+    }
+
+    /** The string "/*" (on Unix). */
+    private static final String FILESEP_STAR = File.separator + "*";
+
+    /**
+     * Given a path element that might be a wildcard, return a list of the elements it expands to.
+     * If the element isn't a wildcard, return a singleton list containing the argument.
+     */
+    private List<String> expandWildcards(String pathElement) {
+        if (pathElement.equals("*")) {
+            return jarFiles(".");
+        } else if (pathElement.endsWith(FILESEP_STAR)) {
+            return jarFiles(pathElement.substring(0, pathElement.length() - 1));
+        } else if (pathElement.equals("")) {
+            return Collections.<String>emptyList();
```

Comment:
The explicit type argument isn't needed.

Reply:
I tried without the explicit type argument, and I got a compiler error.

Reply:
That is strange. #2086 removes the type argument and builds successfully.

Reply:
Weird.  I have no explanation for that.  I only added the type argument because it was needed (locally).  Feel free to remove it.  Thanks for investigating.